### PR TITLE
[test_mtls] Skip test_mtls_enterprise_hsm if not present

### PR DESCRIPTION
**Pull request by lluiscampos**
For QEMU clients build for zeus or older, the SoftHSM will not be present
so skip this test.

### CODE REVIEW

File: tests/tests/test_mtls.py [hunk 1]
```diff
@@ -260,20 +254,28 @@ def test_mtls_enterprise(self, setup_ent_mtls, algorithm):
...
+        out = setup_ent_mtls.device.run(
             "export SOFTHSM2_CONF=/softhsm/softhsm2.conf; mender -show-artifact"
         ).strip()
         assert out == "mtls-artifact"
 
     @MenderTesting.fast
     @pytest.mark.parametrize("algorithm", ["rsa"])
     def test_mtls_enterprise_hsm(self, setup_ent_mtls, algorithm):
+
+        # Check client has SoftHSM (from yocto dunfell forward)
```
Comment (merlin-northern) [inline tests/tests/test_mtls.py:266]: nitpick 2: "Check _if_ the client has .."

File: tests/tests/test_mtls.py [hunk 2]
```diff
@@ -216,28 +212,26 @@ def common_test_mtls_enterprise(
...
-        device.run("systemctl start mender-client")
-        return device
+        env.device.run("systemctl start mender-client")
 
     @MenderTesting.fast
     @pytest.mark.parametrize("algorithm", ["rsa", "ec256", "ed25519"])
     def test_mtls_enterprise(self, setup_ent_mtls, algorithm):
-        # set up the mTLS test environment and return the device
-        device = self.common_test_mtls_enterprise(setup_ent_mtls, algorithm)
+        self.common_test_mtls_enterprise(setup_ent_mtls, algorithm, use_hsm=False)
```
[the remark this answers is not in the document]
Comment (lluiscampos) [reply tests/tests/test_mtls.py:234]: I know, but I thought for the test would be nice to have it explicit: it makes it super clear which is the only test that uses hsm. I can revert if you insist.

**Review by merlin-northern — APPROVED** [review]
lgtm.

File: tests/tests/test_mtls.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (merlin-northern) [reply tests/tests/test_mtls.py:234]: are you crazy? it is ok! :)

**Pull request merged**